v2/pkg/delete/delete_images.go

```diff
@@ -0,0 +1,297 @@
+package delete
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/openshift/oc-mirror/v2/pkg/api/v1alpha2"
+	"github.com/openshift/oc-mirror/v2/pkg/api/v1alpha3"
+	"github.com/openshift/oc-mirror/v2/pkg/archive"
+	"github.com/openshift/oc-mirror/v2/pkg/batch"
+	"github.com/openshift/oc-mirror/v2/pkg/image"
+	clog "github.com/openshift/oc-mirror/v2/pkg/log"
+	"github.com/openshift/oc-mirror/v2/pkg/manifest"
+	"github.com/openshift/oc-mirror/v2/pkg/mirror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+)
+
+type DeleteImages struct {
+	Log              clog.PluggableLoggerInterface
+	Opts             mirror.CopyOptions
+	Batch            batch.BatchInterface
+	Blobs            archive.BlobsGatherer
+	Config           v1alpha2.ImageSetConfiguration
+	Manifest         manifest.ManifestInterface
+	LocalStorageDisk string
+	LocalStorageFQDN string
+}
+
+// DeleteCacheBlobs - does what it says ;)
+// for now we only report errors, it is not critical if the delete cache fails
+// the cache can always be restored (refer to docs/design/v2/delete-functionality.md)
+func (o DeleteImages) DeleteCacheBlobs(ctx context.Context, images []v1alpha3.CopyImageSchema) error {
```

Comment:
I imagined we would do the same as for a registry: call the DELETE HTTP on the cache registry, and let the registry handle the garbage collection of the blobs. 
This sounds pretty risky

Reply:
@sherine-k - this is what I was referring to about using the garbage collector,   its risky using it, why do you want the extra http overhead when you can access the blobs directly. Also the garbage collector  will need to run in a separate thread, with its own quirks to stop (interrupt) the juice is not worth the squeeze also the tests I did deleted all untagged images, for me this was extremely risky.

Reply:
@lmzuccarelli one question, if we delete blobs that are common between two different release versions (or catalog versions) from our cache, is the cache going to become broken? 

Example: 

redhat-operator-index:v4.14 contains blob abc
redhat-operator-index:v4.15 also contains blob abc

I deleted v4.14 from our cache (so abc is gone) and later I want to mirror v4.15 (diskToMirror), is it going to work or the cache became broken?

Reply:
@aguidirh  @sherine-k , I think we need to discuss this i.e maybe not even delete the cache at all, as @aguidirh pointed out it will break the functionality.